clustering/web/WFLY-11898_Session_Attribute_Groups.adoc

```diff
@@ -0,0 +1,74 @@
+= Distributed session attribute groups
+:author:            Paul Ferraro
+:email:             paul.ferraro@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Distributed web sessions support 2 levels of granularity, by which session attributes are mapped to a distributed/remote cache: SESSION and ATTRIBUTE.
+SESSION granularity maps all attributes to a single cache entry, whereas ATTRIBUTE granularity maps each session attribute to a separate cache entry.
+For most applications, ATTRIBUTE granularity is far more efficient than SESSION granularity, since the former only requires the persistence/replication of those attributes that were added/modified/removed within a given request.
+However, if any attributes share object references (e.g. parent/child relationships, foreign keys, etc.) they must use SESSION granularity to ensure that these shared object references are preserved following serialization/deserialization.
+
+This feature request adds a mechanism for users to annotate session attributes into groups, such that users can leverage the efficiency for ATTRIBUTE granularity, while allowing specific attributes to be grouped together into a single cache entry.
```

Comment:
@pferraro what is the behavior if granularity is set to SESSION and some classes are annotated? annotations get ignored?

Reply:
When using SESSION granularity, *all* session attributes shared the same group - so this annotation is technically still respected, but is generally redundant.

Reply:
@pferraro one last clarification: the annotation is e.g. `@Group(name="MY_GROUP_ONE")` hence, when forcing SESSION granularity, the content of attribute name gets ignored, correct?

Reply:
Correct. You can think of session granularity as a generalization of attribute granularity, where all session attributes belong to the same logical group.